sklearn/plot/_gridsearch_results.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+from sklearn.plot import plot_heatmap
+
+
+def plot_gridsearch_results(cv_results, metric='mean_test_score',
+                            xlabel=None, ylabel=None,
+                            title='Grid Search Results', cmap=None,
+                            vmin=None, vmax=None, ax=None, fmt="{:.2f}",
+                            xtickrotation=45, norm=None):
+    """Plot the grid search results as a line chart for 1D search and heatmap
+    for a 2D search. This function will not work if grid-search has more than
+    2 parameters in the search space.
+
+    Parameters
+    ----------
+    cv_results : dict of numpy (masked) ndarrays
+        The cv_results_ attribute of the GridSearchCV object.
+
+    xlabel : string, default=None
+        Label for the x-axis. If None, the first key of the param_grid will
+        be used as the xlabel.
+
+    ylabel : string, default=None
+        Label for the y-axis. If None, the second key of the param_grid will
+        be used as the ylabel.
+
+    metric : string, default="mean_test_score"
+        The metric from the GridSearchCV results to display. This is ignored
+        if only 1 parameter is used in grid search.
+
+    title : string, default="Grid Search Results"
+        Title for the heatmap.
+
+    cmap : string or colormap
+        Matpotlib colormap to use.
+
+    vmin : int, float or None
+        Minimum clipping value. This argument will be passed on to the
+        pcolormesh function from matplotlib used to generate the heatmap.
+
+    vmax : int, float or None
+        Maximum clipping value. This argument will be passed on to the
+        pcolormesh function from matplotlib used to generate the heatmap.
+
+    ax : axes object or None
+        Matplotlib axes object to plot into. If None, the current axes are
+        used.
+
+    fmt : string, default="{:.2f}"
+        Format string to convert value to text.
+
+    xtickrotation : float, default=45
+        Rotation of the xticklabels.
+
+    norm : matplotlib normalizer
+        Normalizer passed to pcolormesh function from matplotlib used to
+        generate the heatmap.
+    """
+
+    import matplotlib.pyplot as plt
+
+    params = sorted(cv_results['params'][0].keys())
+    nparams = len(params)
+
+    if nparams == 1:
+        param = params[0]
+        param_range = sorted(cv_results['param_%s' % param])
+        train_scores_mean = cv_results['mean_train_score']
+        train_scores_std = cv_results['std_train_score']
+        test_scores_mean = cv_results['mean_test_score']
+        test_scores_std = cv_results['std_test_score']
+
+        lw = 2
+        plt.semilogx(param_range, train_scores_mean,
```

Comment:
what i meant is that if the parameter values are say (1, 5, 7, 15) then even if we space them equally on the x-axis, as u suggested using `range(len(train_scores_mean))`, it should be fine because the score will anyways not vary linearly between them and we're just comparing these points as discrete values.